### ISSUE TITLE
WASM target build error. Undeclared symbol "__rcp_fast_varying_double"

### ISSUE THREAD

**Issue by DVLP** [ispc/ispc#2248]
When building with default target the code is working but when switching to `--target=wasm-i32x4`
I'm getting:

#6 0.404 stdlib.ispc:2152:16: Error: Undeclared symbol "__rcp_fast_varying_double".
#6 0.404 stdlib.ispc:2170:12: Error: Undeclared symbol "__rcp_varying_half". Did you
#6 0.404         mean "__exp_varying_half"?

`m_hasRcpd` is false for wasm-i32x4

Could it be related to this issue? Some placeholder declaration missing?
https://github.com/ispc/ispc/issues/553

**Comment by DVLP** [comment]
Ok so building with  -DXE_ENABLED=FALSE seems to be a workaround but it's not good enough if someone is using a pre-built version. i.e. version 18dev has both WASM and XE enabled

Then there's another one in line std 1548 about non-void function that should return a value. I removed the condition and left just `return (float16)(rcp((float)v))` and it worked. Interesting. Could it be that this if/else statement is being interpreted incorrectly by static type checking due to lack of brackets?
```
if (__is_xe_target)
        return __rcp_varying_half(v);
    else
        return (float16)(rcp((float)v));
`

**Comment by dbabokin** [comment]
Seems placeholder declarations are missing in `builtins/target-wasm-i32x4.ll`.

[rcph_decl()](https://github.com/ispc/ispc/blob/575929afa9e9ae0cc44e766b385bf1d70f7dc938/builtins/util.m4#L8104) macro call and some other should be in `builtins/target-wasm-i32x4.ll`.

Note that WASM is not fully functional target yet. @aschrein did great job bring this target in, but switched his focus before completing it. If there are any interest in this target, I would fully support contributions to make it work.

**Comment by Lectem** [comment]
I tried to fix this earlier based on the main branch.
For me, only `__rcp_fast_varying_double` was missing. It seems there is an implementation in the ll file for `__rcp_varying_double` but not the fast one. (the float version just uses the same code for both).
I tried doing the same for the double version and could generate my library correctly. However it seems that the node.js version shipped with latest emscripten won't run it.
Apparently there is an unknown Opcode generated which I find surprising. (it is probably unrelated to this function).
Is there any way to check the output of ispc for errors, perhaps by using a disassembler? Any guidance would be helpful.
